drivers/ethernet/eth_stellaris.c

```diff
@@ -0,0 +1,399 @@
+	frame_len = eth_stellaris_rx_pkt(dev, pkt);
+	if (frame_len < 0) {
+		LOG_ERR("Failed to append data to buffer");
+		goto pkt_unref;
+	}
+
+	ret = net_recv_data(iface, pkt);
+	if (ret < 0) {
+		LOG_ERR("Failed to place frame in RX Queue");
+		goto pkt_unref;
+	}
+
+	if (IS_ENABLED(CONFIG_NET_STATISTICS_ETHERNET)) {
+		struct net_eth_hdr *pkt_hdr;
+
+		/* Update statistics counters */
+		eth_stats_update_bytes_rx(iface, frame_len - 6);
+		eth_stats_update_pkts_rx(iface);
+		pkt_hdr = NET_ETH_HDR(pkt);
+		if (net_eth_is_addr_broadcast(&pkt_hdr->dst)) {
+			eth_stats_update_broadcast_rx(iface);
+		} else if (net_eth_is_addr_multicast(&pkt_hdr->dst)) {
+			eth_stats_update_multicast_rx(iface);
+		}
+	}
+
+	return;
+
+pkt_unref:
+	net_pkt_unref(pkt);
+
+err_mem:
+	eth_stellaris_rx_error(iface);
+}
+
+static void eth_stellaris_isr(void *arg)
+{
+	/* Read the interrupt status */
+	struct device *dev = (struct device *)arg;
+	struct eth_stellaris_runtime *dev_data = DEV_DATA(dev);
+	int isr_val = sys_read32(REG_MACRIS);
+	u32_t lock;
+
+	lock = irq_lock();
+
+	/* Acknowledge the interrupt. */
+	sys_write32(isr_val, REG_MACRIS);
+
+	if (isr_val & BIT_MACRIS_RXINT) {
+		eth_stellaris_rx(dev);
+	}
+
+	if (isr_val & BIT_MACRIS_TXEMP) {
+		dev_data->tx_err = false;
+		k_sem_give(&dev_data->tx_sem);
+	}
+
+	if (isr_val & BIT_MACRIS_TXER) {
+		LOG_ERR("Transmit Frame Error");
+		eth_stats_update_errors_tx(dev_data->iface);
+		dev_data->tx_err = true;
+		k_sem_give(&dev_data->tx_sem);
+	}
+
+	if (isr_val & BIT_MACRIS_RXER) {
+		LOG_ERR("Error Receiving Frame");
+		eth_stellaris_rx_error(dev_data->iface);
+	}
+
+	if (isr_val & BIT_MACRIS_FOV) {
+		LOG_ERR("Error Rx Overrun");
+		eth_stellaris_rx_error(dev_data->iface);
+	}
+
+	irq_unlock(lock);
+}
+
+static void eth_stellaris_init(struct net_if *iface)
+{
+	struct device *dev = net_if_get_device(iface);
+	struct eth_stellaris_config *dev_conf = DEV_CFG(dev);
+	struct eth_stellaris_runtime *dev_data = DEV_DATA(dev);
+
+	dev_data->iface = iface;
+
+	/* Assign link local address. */
+	net_if_set_link_addr(iface,
+			     dev_data->mac_addr, 6, NET_LINK_ETHERNET);
+
+	ethernet_init(iface);
+
+	/* Initialize semaphore. */
+	k_sem_init(&dev_data->tx_sem, 0, 1);
+
+	/* Initialize Interrupts. */
+	dev_conf->config_func(dev);
+}
+
+static struct net_stats_eth *eth_stellaris_stats(struct device *dev)
+{
+	return &(DEV_DATA(dev)->stats);
+}
+
+static int eth_stellaris_dev_init(struct device *dev)
+{
+	u32_t value;
+
+	/* Assign MAC address to Hardware */
+	eth_stellaris_assign_mac(dev);
+
+	/* Program MCRCTL to clear RXFIFO */
+	value = BIT_MACRCTL_RSTFIFO;
+	sys_write32(value, REG_MACRCTL);
+
+	/* Enable transmitter */
+	value = BIT_MACTCTL_DUPLEX | BIT_MACTCTL_CRC |
+		BIT_MACTCTL_PADEN | BIT_MACTCTL_TXEN;
+	sys_write32(value, REG_MACTCTL);
+
+	/* Enable Receiver */
+	value = BIT_MACRCTL_BADCRC | BIT_MACRCTL_RXEN;
+	sys_write32(value, REG_MACRCTL);
+
+	return 0;
+}
+
+static struct device DEVICE_NAME_GET(eth_stellaris);
```

Comment:
You should not need this

Reply:
But didn't we agree to this: https://github.com/zephyrproject-rtos/zephyr/pull/10097#discussion_r230334164

Reply:
Ah yes, my bad. I am used to other type of drivers (spi, gpio...) and there the irq config function comes after the DEVICE_INIT() macro call. Nevermind.